target/snitch_cluster/sw/snax/gemmx/src/snax-gemmx-lib.c

```diff
@@ -204,10 +188,24 @@ void set_gemmx_csr(int tempLoop0, int tempLoop1, int tempLoop2,
     // set the constants for the SIMD unit
     csrw_ss(SIMD_CSR0, csr0);
     csrw_ss(SIMD_CSR1, csr1);
-    csrw_ss(SIMD_CSR2, csr2);
+
+    // set the shared bitpacked shift
+    csrw_ss(SIMD_SHARED_BITPACKED_SHIFT0, shared_bitpacked_shift0);
+    csrw_ss(SIMD_SHARED_BITPACKED_SHIFT1, shared_bitpacked_shift1);
+
+    // set the shared multipliers
+    csrw_ss(SIMD_SHARED_MULTIPLIER0, shared_multiplier0);
+    csrw_ss(SIMD_SHARED_MULTIPLIER1, shared_multiplier1);
+    csrw_ss(SIMD_SHARED_MULTIPLIER2, shared_multiplier2);
+    csrw_ss(SIMD_SHARED_MULTIPLIER3, shared_multiplier3);
+    csrw_ss(SIMD_SHARED_MULTIPLIER4, shared_multiplier4);
+    csrw_ss(SIMD_SHARED_MULTIPLIER5, shared_multiplier5);
+    csrw_ss(SIMD_SHARED_MULTIPLIER6, shared_multiplier6);
+    csrw_ss(SIMD_SHARED_MULTIPLIER7, shared_multiplier7);
```

Comment:
LOL now there are much more CSRs to configure. As more features are added, we are getting close to NVDLA registers hahahaha.

Reply:
You are right. There are a lot of control registers needed especially if we want to support per channel quantization. We can retrain the models with per tensor quantization but there might be some accuracy lost.